### ISSUE TITLE
JSDoc @import not detecting use in @callback

### ISSUE THREAD

**Issue by turadg** [microsoft/TypeScript#58368]
### 🔎 Search Terms

jsdoc import @import callback lsp

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because it's a new feature in 5.5. It was appearing in nightlies and I waited for the beta to confirm it's still there.

### 💻 Code

```js
/**
 * @import {Checker, Passable} from '@endo/pass-style';
 */

/**
 * @callback GetChecker
 * @param {Passable} payload
 * @returns {Checker}
 */

/**
 * @typedef {object} Misc
 * @property {Passable} passable
 */
```

### 🙁 Actual behavior

The `Checker` import is dim in an IDE and hovering says,
```
'Checker' is declared but never used.ts(6196)
```

### 🙂 Expected behavior

LSP detects that `Checker` is in the return of `GetChecker`. 

### Additional information about the issue

The use is detected in `@typedef` and `@type`. E.g. as with `Passable` in the example code.

**Comment by a-tarasyuk** [comment]
I attempted to reproduce this issue, but it appears to be working correctly. Could you please provide additional details?

<img width="844" alt="Screenshot 2024-05-01 at 01 18 40" src="https://github.com/microsoft/TypeScript/assets/509265/5a755b84-6852-415c-b8de-9a0e95fe34a7">

**Comment by turadg** [comment]
Thanks for trying. Looking into it further, in my case it's reporting `ts(6133)` until I navigate to the definition. It resets again to thinking it's unused when I restart TS.

[Here is a video](https://drive.google.com/file/d/1jc1IlV-z5ob3Dm7Q-cyBwTy8lxkEONhy/view?usp=sharing) demonstrating in VS Code. 
```
Version: 1.88.1
Commit: e170252f762678dec6ca2cc69aba1570769a5d39
Date: 2024-04-10T17:43:08.196Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Darwin arm64 23.4.0
```

**Comment by turadg** [comment]
More info: I think this only happens when the type is available by ambients. Then the imported type reports that it's not used.

This may not be a bug per current spec but it is a bit of a UX hazard because ambient resolution can differ by build order etc. So a type that's "not used" at coding time may be necessary at build time. The message that says it's not used suggests it can be removed, but that breaks the build.

**Comment by RyanCavanaugh** [comment]
Looks like a bug, but I can't repro this either. We can investigate if there's a way to reproduce it

**Comment by michaeljaltamirano** [comment]
@RyanCavanaugh I am having trouble putting together a reproduction, but the details of my case somewhat match up with this issue. However, I noticed that if I move the JSDoc import statement to happen _before_ the last non-JSDoc `import`, it removes the `ts(6133)` error:


Before:
<img width="444" alt="Screenshot 2024-06-20 at 4 36 46 PM" src="https://github.com/microsoft/TypeScript/assets/13544620/36af2a7b-67cf-4fc6-a4db-b013be032387">

After:
<img width="408" alt="Screenshot 2024-06-20 at 4 37 15 PM" src="https://github.com/microsoft/TypeScript/assets/13544620/a4e5125d-0e02-4055-9555-9d5fcf706d37">

Hope this helps!

**Comment by remcohaszing** [comment]
Here’s a reproduction: [playground](https://www.typescriptlang.org/play/?filetype=js#code/PQKhCgAIUgBBLAtgBwPYCcAukDeA5VAEwFMBnAX0gDN1VFIByRQgQ1MwahgRQ2xwAKqeADtMAGkhDS8TPFQjKNOowCuI+O07Rg4cKAjRIAcWLZMAC2KRUVKqTM2qkFpGQt0DwpHWbsaUUwAOi4uOHd0FnpBYTFKALEwyEgAFSsfDXY3WOxbSAB3C3gAYwtITFRIAHNHS2tbezMQo1h0M1V0EVJcEVVEACNidHIk1PSGh1znOoy-bMDmkF1iAA80LGp1YrkFarMhQIB5O0mACgTMAEpcKEg2zA6RSAM4TABPZGscXoGhyiXIOcckEJmZLuARvowGFTOZ0uwPFNGlMXG4PF5ZlkLoswrAIlFcAQSBRICIiMRRmlrL4sTknAUiqVypUanDrAiNqDgrj7o9ut8+oNhpT4ZhEU5kfSZjT-MCuMs1nxNiJtvInqyicQAMpirDHZGnMkka44W68zp7TAHMT6s4vWDvT64aSyNX-YCAo3EIJoGQ7ESXIIcq4QvQGGG1dLEETeLn01wRDEy+ZiHEtfHRTUkr0i6mZWWBemFEplCqW8pRmMSyaLODmro9QV-XOQaOxk6OPLS-Mp7k6cCrdbYKhbf2WzUAURjtrMhvJJrN7QtrOtmBnmFO9sdXxd-vdnvJPtQfrVgbb4JGQA)

It’s based on [this commit](https://github.com/mdx-js/mdx-analyzer/commit/bafdc57e39c75c7f494dd8b9ef7b4b1443725fea), where it happens in `packages/language-service/lib/mdast-utils.js` and `packages/language-service/lib/tsconfig.js` in seemingly arbritrary situations.

**Comment by lishaduck** [comment]
I had a file with no value imports, so I messed around with it and found that adding a semicolon or `void 0` fixed it. Probably any statement. Doesn't really help, but for files without imports, try the `void 0` trick.

**Comment by lishaduck** [comment]
Thank you @a-tarasyuk!!! ❤
